Registration/RegisterWithAzure.psm1

```diff
@@ -237,67 +237,175 @@ function Set-AzsRegistration{
-
     Log-Output "*********************** Begin log: $($PSCmdlet.MyInvocation.MyCommand.Name) ***********************`r`n"
+    
+    $stampOutput = Invoke-RegistrationStep1AgainstPEP -PrivilegedEndpointCredential $PrivilegedEndpointCredential `
+                                                      -PrivilegedEndpoint $PrivilegedEndpoint `
+                                                      -AzureContext $AzureContext `
+                                                      -BillingModel $BillingModel `
+                                                      -MarketplaceSyndicationEnabled:$MarketplaceSyndicationEnabled `
+                                                      -UsageReportingEnabled:$UsageReportingEnabled `
+                                                      -AgreementNumber $AgreementNumber ` # required for Capacity billing model
+                                                      -MsAssetTag $MsAssetTag # required for Custom billing model
+
+    $azureOutput = Invoke-RegistrationStep2AgainstAzure -AzureContext $AzureContext `
+                                                        -RegistrationName $RegistrationName `
+                                                        -ResourceGroupName $ResourceGroupName `
+                                                        -ResourceGroupLocation $ResourceGroupLocation `
+                                                        -RegistrationToken $stampOutput.RegistrationToken `
+                                                        -ServicePrincipalObjectId $stampOutput.ServicePrincipalObjectId
+
+    Invoke-RegistrationStep3AgainstPEP -PrivilegedEndpointCredential $PrivilegedEndpointCredential `
+                                       -PrivilegedEndpoint $PrivilegedEndpoint `
+                                       -ActivationKey $azureOutput
+
+    Log-Output "Your environment is now registered and activated using the provided parameters."
+    Log-Output "*********************** End log: $($PSCmdlet.MyInvocation.MyCommand.Name) ***********************`r`n`r`n"
+}
+
+Invoke-RegistrationStep1AgainstPEP{
+[CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [PSCredential] $PrivilegedEndpointCredential,
+
+        [Parameter(Mandatory = $true)]
+        [String] $PrivilegedEndpoint,
+
+        [Parameter(Mandatory = $true)]
+        [PSObject] $AzureContext,
+        
+        [Parameter(Mandatory = $true)]
+        [ValidateSet('Capacity', 'PayAsYouUse', 'Development','Custom')]
+        [string] $BillingModel,
+
+        [Parameter(Mandatory = $false)]
+        [switch] $MarketplaceSyndicationEnabled,
+
+        [Parameter(Mandatory = $false)]
+        [switch] $UsageReportingEnabled,
+
+        [Parameter(Mandatory = $false)]
+        [string] $AgreementNumber,
+
+        [Parameter(Mandatory=$false)]
+        [string] $MsAssetTag
+    )
+
+    if([string]::IsNullOrEmpty($Script:registrationLog)){
```

Comment:
@sarathys and @keystroke any suggestion regarding how to handle registration log file in 3 split functions?
i.e. Invoke-RegistrationStep1AgainstPEP, Invoke-RegistrationStep2AgainstAzure and Invoke-RegistrationStep3AgainstPEP

Reply:
I see. I suppose desired behavior would be that calling the "wrapper" command to do all steps would produce a single log file, whereas when the individual functions are called, they may need separate log files. The logging approach we have here is a little odd, but in the context of how it's setup, it looks like the big wrapper function can have something like:

````powershell
try {
    New-RegistrationLogFile -RegistrationFunction $PSCmdlet.MyInvocation.MyCommand.Name
    # step1
    # step2
    # step3
} finally {
    $Script:registrationLog = $null
}
````

Then you keep the logic you've added to the helper functions checking if the log file has been initialized. The try/finally ensures that any premature exit or aborting of the run won't leave the log file defined and interfere with subsequent commands.

Alternative design might be that all these commands accept an optional parameter of -LogFilePath. If any command has that passed, it traces to that file. If no path is passed-in, then it initializes a log file using current logic. This also allows user to provide custom log file paths if desired. But that is more refactoring than you may want to do in this change.